modules/ROOT/pages/provisioning-kubevirt.adoc

```diff
@@ -0,0 +1,96 @@
+= Provisioning Fedora CoreOS on KubeVirt
+
+This guide shows how to provision new Fedora CoreOS (FCOS) nodes on any KubeVirt-enabled Kubernetes cluster.
+
+== Prerequisites
+
+Before provisioning an FCOS machine, you must have an Ignition configuration file containing your customizations. If you do not have one, see xref:producing-ign.adoc[Producing an Ignition File].
+
+You also need to have access to a Kubernetes environment with https://kubevirt.io/user-guide/operations/installation/[KubeVirt] installed.
+
+== Downloading the KubeVirt Image
+
+Fedora CoreOS is designed to be updated automatically, with different schedules per stream.
+Once you have picked the relevant stream, download the ociarchive:
+
+NOTE: For more information on FCOS stream offerings see xref:update-streams.adoc[Update Streams].
+
+[source, bash]
+----
+STREAM='stable'
+coreos-installer download -s $STREAM -p kubevirt -f ociarchive
+----
+
+Alternatively, you can manually download a KubeVirt image from the
+https://getfedora.org/coreos/download?tab=cloud_operators[download page] and directly download the ociarchive from there.
+
+== Uploading the Image to a Container Registry
+
+.Upload the image to your container registry
+[source, bash]
+----
+FILE=fedora-coreos-XX.XXXXXXXX.X.X-kubevirt.x86_64.ociarchive
+skopeo copy oci-archive:${FILE} docker://quay.io/myorg/fcos:dev
+----
+
+== Launching a VM instance
+
+Now that you have an image it can be used on your VMI definitions.
+
+You'll also need the Ignition config.
+It can be provided directly on the VMI like in this example, or https://kubevirt.io/user-guide/virtual_machines/startup_scripts/#startup-scripts[uploaded to a secret].
+
+.Launching a VM instance
+[source, bash]
+----
+cat << END > vmi.yaml
+---
+apiVersion: kubevirt.io/v1
+kind: VirtualMachineInstance
+metadata:
+  name: my-fcos
+spec:
+  domain:
+    devices:
+      disks:
+      - disk:
+          bus: virtio
+        name: containerdisk
+      - disk:
+          bus: virtio
+        name: cloudinitdisk
+      rng: {}
+    resources:
+      requests:
+        memory: 1024M
+  volumes:
+  - containerDisk:
+      image: quay.io/myorg/fcos:dev
+    name: containerdisk
+  - name: cloudinitdisk
+    cloudInitConfigDrive:
+      userData: |-
+        {
+          "ignition": {
+            "version": "3.3.0"
+          },
+          "passwd": {
+            "users": [
+              {
+                "name": "core",
+                "sshAuthorizedKeys": [
+                  "ssh-rsa AAAAB3NzaC1yc2 [...]"
+                ]
+              }
+            ]
+          }
+        }
```

Comment:
Rather than expecting users to figure out how to inline the Ignition config, let's give them an example.  Since we're already using a here-doc with `cat`, this might be something like:

```suggestion
$(cat config.ign | sed 's/^/        /')
```

though that's a bit awkward, since it hardcodes the indentation level.

Reply:
Switched to using a k8s secret where the ignition file can directly be uploaded via a `kubectl` call. That may be the better solution. :+1: